File: parent327/pinyougou-web-manager/src/main/java/cn/itcast/core/controller/SeckillQueryController.java
package cn.itcast.core.controller;


import cn.itcast.core.pojo.seckill.SeckillOrder;
import cn.itcast.core.service.SeckillQueryService;
import com.alibaba.dubbo.config.annotation.Reference;
import entity.PageResult;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/seckillQuery")
public class SeckillQueryController {

    @Reference
    private SeckillQueryService seckillQueryService;

    //查询分页对象  带条件   $scope.searchEntity = '{name:O,fr..:O}'
    @RequestMapping("/search")
    public PageResult search(Integer pageNum, Integer pageSize, @RequestBody SeckillOrder seckillOrder){
        PageResult pageResult = seckillQueryService.search(pageNum, pageSize, seckillOrder);
        return pageResult;
    }
    //查询一个品牌
    @RequestMapping("/findOne")
    public SeckillOrder findOne(Long id){
        SeckillOrder one = seckillQueryService.findOne(id);
        return one;

    }
}

File: parent327/pinyougou-web-manager/src/main/webapp/js/controller/seckillQueryController.js
// 定义控制器:
app.controller("seckillQueryController",function($scope,$controller,seckillQueryService){
    // AngularJS中的继承:伪继承
    $controller('baseController',{$scope:$scope});


    // 查询一个:
    $scope.findById = function(id){
        seckillQueryService.findOne(id).success(function(response){
            // {id:xx,name:yy,firstChar:zz}
            $scope.entity = response;
        });
    }


    $scope.searchEntity={};

    // 假设定义一个查询的实体：searchEntity
    $scope.search = function(page,rows){
        // 向后台发送请求获取数据:
        seckillQueryService.search(page,rows,$scope.searchEntity).success(function(response){
            $scope.paginationConf.totalItems = response.total;
            $scope.list = response.rows;
        });
    }

});
